### DISCUSSION TITLE
"Group tasks by app" should still allow dragging of entire groups left/right (custom order)

### DISCUSSION THREAD

**Discussion by derek-cb** [lawand-dot-io/taskbar#124] [Feature requests]
Currently, "Group tasks by app" only allows for chronological ordering of groups. BUT in Windows, I'm used to being allowed to drag an entire group left/right to set my own custom order. Would love for this feature to come to Taskbar, as it's a major pain point for me having to close/re-open each app to get the order I want currently.

**Comment by lawand-dot-io** [comment]
Hello. Yeah I understand what you mean. When I released the first version of Taskbar, window grouping wasn't supported. At the moment window grouping is still basic and I can add more features like dragging and dropping whole groups or dragging windows within groups. Will keep this in mind and as always, features that are the most upvoted and the most likely to be implemented first. Cheers

**Comment by lawand-dot-io** [comment]
By the way, for the moment, wouldn't it be easier for you to disable window grouping and drag and drop windows to the positions you want, and then you only need to drag and drop new windows to the correct position. That might be easier than closing and opening apps.

**Reply by derek-cb** [reply]
This workaround would definitely work for me if [#130](https://github.com/lawand-dot-io/taskbar/discussions/130) were fixed, but without it, I keep having to drag my existing window back to its former location after exiting full-screen mode. Grouping tasks by app fixes the issue described in [#130](https://github.com/lawand-dot-io/taskbar/discussions/130), but then I can't re-order the groups themselves without having to quit the app entirely (since currently, app grouping is chronologically-based).

Alternatively, if this (#124) issue were fixed, I wouldn't need [#130](https://github.com/lawand-dot-io/taskbar/discussions/130) to be fixed, as I could retain my app order after exiting full-screen mode, while also having my preferred order of app groups (via drag-and-drop).

Let me know if that makes sense!
Thank you

**Reply by lawand-dot-io** [reply]
@derek-cb thanks for the input. I understand your use case. I would say it's more likely that I implement dragging entire groups than fix the issue with fullscreen windows since that would be a bit complicated. In any case I will keep all those issues in mind for when I have time to implement those features. At the moment I am implementing an option for Taskbar size and after that I will work on multi-monitor support and hopefully not too much later I would allow dragging entire groups. Cheers

**Comment by lawand-dot-io** [comment]
Closing this request since I realized there's actually an older duplicate (https://github.com/lawand-dot-io/taskbar/discussions/77). Will keep this post's discussion in mind when looking at the original request.
